Source/Dafny/Resolver.cs

```diff
@@ -5497,7 +5497,7 @@ public class XConstraint {
           return true;
         }
 
-        resolver.ConstrainSubtypeRelation(collectionElementType, requestedVariableType.GetCompilableParentType(), errorMsg, true);
+        resolver.ConstrainSubtypeRelation(requestedVariableType.GetCompilableParentType(), collectionElementType, errorMsg, true);
```

Comment:
Should a test be added for this change as well?

Reply:
I understand your concern. If the code still works after swapping two arguments, but this version is more meaningful, we should ensure tests break if we revert that change (or if someone comes after me thinking they will fix a potential bug).
As of today though, I don't see an easy way to test this.
To make a test that would work on the second but fail on the first
- We would need a case where the bound variable's type is not explicitly given in the comprehension. Otherwise, this constraint would not even be emitted in the first place.
- We would also need that, at the time of confirming this constraint
  - the bound variable's type that must hold after the range is determined (first "if" of "ConfirmSubsetTypeOfCompilable" in Resolver.cs)
  - the collection's type must not be determined yet (second "if"  of "ConfirmSubsetTypeOfCompilable" in Resolver.cs). That "if" could also be a point of discussion, basically, it says that if the two types are determined, then we can just let the verifier do its job, we don't need resolution checks, even if the types would not match.

Given the above, and that the confirmation of the constraint happens after the comprehension is resolved, I don't know how to create a test where the collection's type is determined by the inferred type of the bound variable. I even discovered a bug doing so (#2074 - thanks for your prompt !) so it's unlikely I can myself even create a test for that purpose anytime soon.